src/Integration.Vsix.UnitTests/SonarLintTagger/AccumulatingIssueConsumerTests.cs

```diff
@@ -0,0 +1,182 @@
+
+            using (new AssertIgnoreScope())
+            {
+                testSubject.Accept(ValidFilePath, issues);
+            }
+
+            callbackSpy.CallCount.Should().Be(1);
+            if (isMappableToSnapshot)
+            {
+                callbackSpy.LastSuppliedIssues.Should().BeEquivalentTo(issues);
+            }
+            else
+            {
+                callbackSpy.LastSuppliedMarkers.Should().BeEmpty();
+            }
+        }
+
+        [TestMethod]
+        public void Accept_MultipleCallsToAccept_IssuesAreAccumulated()
+        {
+            var callbackSpy = new OnIssuesChangedCallbackSpy();
+            var firstSetOfIssues = new[]
+            {
+                CreateIssue(1, 1), CreateIssue(2, 2)
+            };
+
+            var secondSetOfIssues = new[]
+            {
+                CreateIssue(3,3), CreateIssue(4,4)
+            };
+
+            var snapshot = CreateSnapshot(lineCount: 10);
+            var converter = CreateWrapAllConverter();
+
+            var testSubject = new AccumulatingIssueConsumer(snapshot, ValidFilePath, callbackSpy.Callback, converter);
+
+            // 1. First call
+            testSubject.Accept(ValidFilePath, firstSetOfIssues);
+
+            callbackSpy.CallCount.Should().Be(1);
+            callbackSpy.LastSuppliedIssues.Should().BeEquivalentTo(firstSetOfIssues);
+
+            // 2. Second call
+            testSubject.Accept(ValidFilePath, secondSetOfIssues);
+
+            callbackSpy.CallCount.Should().Be(2);
+            callbackSpy.LastSuppliedIssues.Should().BeEquivalentTo(firstSetOfIssues.Union(secondSetOfIssues));
+        }
+
+        private class OnIssuesChangedCallbackSpy
+        {
+            public int CallCount { get; private set; }
+            public IList<IssueMarker> LastSuppliedMarkers { get; private set; }
+            public IList<IAnalysisIssue> LastSuppliedIssues
+            {
+                get
+                {
+                    return LastSuppliedMarkers?.Select(x => x.Issue).ToList();
+                }
+            }
+
+            public void Callback(IEnumerable<IssueMarker> issueMarkers)
+            {
+                CallCount++;
+                LastSuppliedMarkers = issueMarkers?.ToList();
+            }
+        }
+
+        private static ITextSnapshot CreateSnapshot(int lineCount)
+        {
+            var mockSnapshot = new Mock<ITextSnapshot>();
+            mockSnapshot.Setup(x => x.LineCount).Returns(lineCount);
+            return mockSnapshot.Object;
+        }
+
+        private static IAnalysisIssue CreateIssue(int startLine, int endline) =>
+            new DummyAnalysisIssue
+            {
+                StartLine = startLine,
+                EndLine = endline,
+                Message = "any message"
+            };
+
+        private static IIssueToIssueMarkerConverter CreateWrapAllConverter()
```

Comment:
Maybe CreateCallThroughConverter? Or CreateDummyConverter/ just CreateIssueConverter. I didn't understand what `WrapAll` meant and had to look at the code anyway.

Reply:
I've changed it to `CreatePassthroughConverter`